java/vector/src/main/java/org/apache/arrow/vector/complex/RunEndEncodedVector.java

```diff
@@ -312,8 +298,68 @@ public TransferPair getTransferPair(Field field, BufferAllocator allocator, Call
    */
   @Override
   public TransferPair makeTransferPair(ValueVector target) {
-    throw new UnsupportedOperationException(
-        "RunEndEncodedVector does not support makeTransferPair(ValueVector)");
+    return new TransferImpl((RunEndEncodedVector) target);
+  }
+
+  private class TransferImpl implements TransferPair {
+
+    RunEndEncodedVector to;
+    TransferPair dataTransferPair;
+    TransferPair reeTransferPair;
+
+    public TransferImpl(String name, BufferAllocator allocator, CallBack callBack) {
+      this(new RunEndEncodedVector(name, allocator, field.getFieldType(), callBack));
+    }
+
+    public TransferImpl(Field field, BufferAllocator allocator, CallBack callBack) {
+      this(new RunEndEncodedVector(field, allocator, callBack));
+    }
+
+    public TransferImpl(RunEndEncodedVector to) {
+      this.to = to;
+      if (to.getRunEndsVector() instanceof ZeroVector) {
+        to.initializeChildrenFromFields(field.getChildren());
+      }
+      reeTransferPair = getRunEndsVector().makeTransferPair(to.getRunEndsVector());
+      dataTransferPair = getValuesVector().makeTransferPair(to.getValuesVector());
+    }
+
+    /**
+     * Transfer this vector'data to another vector. The memory associated with this vector is
+     * transferred to the allocator of target vector for accounting and management purposes.
+     */
+    @Override
+    public void transfer() {
+      to.clear();
+      dataTransferPair.transfer();
+      reeTransferPair.transfer();
+      if (valueCount > 0) {
+        to.setValueCount(valueCount);
+      }
+      clear();
+    }
+
+    /**
+     * Slice this vector at desired index and length and transfer the corresponding data to the
+     * target vector.
+     *
+     * @param startIndex start position of the split in source vector.
+     * @param length length of the split.
+     */
+    @Override
+    public void splitAndTransfer(int startIndex, int length) {
+      throw new UnsupportedOperationException();
```

Comment:
Do we have an issue filed for this?

Reply:
I don't think we have, but we should generally complete split and transfer features before we finalise the C Data interface. IMHO it is important to cover that in this stage.

Reply:
At least this was the approach we took for StringView and ListView before.

Reply:
I have only implemented the necessary functions to pass through RoundTripTest. I'll implement it in this PR, if you all think so.

Reply:
@ViggoC if this requires a lot more code or time, let's do it later, it is not a must. But later means as an immediate next feature. I was merely mentioning my practice.

Reply:
I think it's not time consuming, I'll implement it here.

Reply:
Thanks. I'll wait for that.